<?xml version="1.0" encoding="utf-8"?>

<launch>
  
  <!-- Use YOLOv4 -->
  <arg name="image" default="usb_cam/image_raw" />

  <!-- Console launch prefix -->
  <arg name="launch_prefix" default=""/>

  <!-- Config and weights folder. -->
  <arg name="yolo_weights_path" default="$(find darknet_ros)/yolo_network_config/weights"/>
  <arg name="yolo_config_path"  default="$(find darknet_ros)/yolo_network_config/cfg"/>

  <!-- ROS and network parameter files -->
  <arg name="ros_param_file_1"     default="$(find darknet_ros)/config/ros_1.yaml"/>
  <arg name="network_param_file_1" default="$(find darknet_ros)/config/delivery_v4.yaml"/>
  <arg name="ros_param_file_2"     default="$(find darknet_ros)/config/ros_2.yaml"/>
  <arg name="network_param_file_2" default="$(find darknet_ros)/config/qualify_ver2.yaml"/>

  <!-- Load parameters -->
  <rosparam command="load" ns="camera1/darknet_ros" file="$(arg ros_param_file_1)"/>
  <rosparam command="load" ns="camera1/darknet_ros" file="$(arg network_param_file_1)"/>
  <rosparam command="load" ns="camera2/darknet_ros" file="$(arg ros_param_file_2)"/>
  <rosparam command="load" ns="camera2/darknet_ros" file="$(arg network_param_file_2)"/>

  <!-- Start darknet and ros wrapper -->
  <node ns="camera1" pkg="darknet_ros" type="darknet_ros" name="darknet_ros" output="screen" launch-prefix="$(arg launch_prefix)">
    <param name="weights_path"          value="$(arg yolo_weights_path)" />
    <param name="config_path"           value="$(arg yolo_config_path)" />
    <remap from="/usb_cam/image_raw"  to="$(arg image)" />
  </node>
  <node ns="camera2" pkg="darknet_ros" type="darknet_ros" name="darknet_ros" output="screen" launch-prefix="$(arg launch_prefix)">
    <param name="weights_path"          value="$(arg yolo_weights_path)" />
    <param name="config_path"           value="$(arg yolo_config_path)" />
    <remap from="/usb_cam/image_raw"  to="$(arg image)" />
  </node>

 <!--<node name="republish" type="republish" pkg="image_transport" output="screen" 	args="compressed in:=/front_camera/image_raw raw out:=/camera/image_raw" /> -->

<!-- send class -->
  <!--<node ns="camera1" pkg="darknet_ros" type="identify.py" name="obj_ID" output="screen"> 
  </node> -->

</launch>
